app/views/projects/show.haml

```diff
@@ -121,6 +121,10 @@
   %p The following import overrides are in effect for this project.
   .mb-4.card
     = render 'hmis_csv_importer/import_overrides/table', overrides: overrides, data_source: @project.data_source, editable: false, show_associated_project: false
+
+- if RailsDrivers.loaded.include?(:client_location_history) && can_view_project_locations?
```

Comment:
I need to make sure I'm implementing this permission correctly; I realized it's the first time I have tried to work with granular permissions in the warehouse and it seems it's set up a bit differently. Are there separate Collections and User Groups in WH, or does WH use the same ones HMIS uses?

Reply:
The warehouse has separate collections and user groups.  This will check globally if the user can see any project locations.  It's probably worth checking if the user can `can_view_project_locations?` for the given project.  It's Friday, so I'm probably missing it, but I'm not sure we have a method for that yet.  Essentially we need to call:
```
GrdaWarehouse::Hud::Project.viewable_by(current_user, permission: :can_view_project_locations).where(id: @project.id).exists?
```

Reply:
If you want to turn that into a method on Project:
```
def can?(user, permission: :can_view_projects)
  self.class.viewable_by(user, permission: permission).where(id: id).exists?
end
```

I wouldn't be opposed.  Then the check could be `@project.can?(current_user, permission: :can_view_project_locations)` which feels a bit cleaner.

Reply:
Sounds good. I wonder if it's worth trying to more closely match the hmis pattern for this, which I think lives here in this `#{permission}_for?` helper: https://github.com/greenriver/hmis-warehouse/blob/e28ca6f6c13632c733079bcb515ca4904539d36a/drivers/hmis/app/models/hmis/user.rb#L134-L137

Reply:
But maybe that'd just result in a lot more code duplication, like duplicating the `EntityAccessLoaderFactory` https://github.com/greenriver/hmis-warehouse/blob/e28ca6f6c13632c733079bcb515ca4904539d36a/drivers/hmis/app/models/hmis/entity_access_loader_factory.rb#L24

Reply:
Hmm, I'm not opposed to a more generic solution, but I'm liking the simplicity of a `can?` method. With `permission_for?` we'd need to define similar methods on each entity (I think that's kinda-sorta what EntityAccessLoaderFactory is going.  An alternative route, that's probably more inline with where we've been going is to make a class that arbitrates this.  So you could call something like:

`PermissionProvider.new(user, entity).can?(permission)`

Then we could encapsulate the calculations for what's possible for each entity in there.   That potentially allows us to do some batch calculations as well.

I'm not sure I have the clock cycles to devote to thinking through that whole thing right now, but if you think that's a better approach, we could spend some time on it in the near future.

Reply:
It seems like maybe this is no longer necessary, with the location map moved to its own page? However, see https://github.com/greenriver/hmis-warehouse/pull/4696#discussion_r1763250524

Reply:
I agree, if it's on its own page, this should require access to the project and access to the new permission.  Ideally, we'd correlate those two to make sure you have access to _this_ project's locations (for ACL users) but probably fine to require each of them for now.